### PR TITLE
CL-731 Add new custom element options (stopThreshold, new imageCompareFormat)

### PR DESCRIPTION
**Pull request by adi-wan-askui**

### CODE REVIEW

File: packages/askui-nodejs/src/core/model/custom-element-json.ts [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
+ * It allows filtering for a UI element based on an image instead of using
+ * text or element descriptions like `button().withText('Submit')` in
+ * `await aui.click().button().withText('Submit').exec()`.
  *
  * **Important:** The `CustomElementJson` needs to capture as accurately as possible
  * what the custom element looks like during test execution as otherwise
  * our machine learning models cannot find it, even with the additional data
  * provided. This is especially true for the resolution used while cropping
- * the `CustomElementJson.customImage` which should match the resolution during
+ * the `CustomElementJson.customImage` which should be close the resolution during
```
Comment (JohannesDienst-askui) [inline packages/askui-nodejs/src/core/model/custom-element-json.ts:12]: the `CustomElementJson.customImage` which should be close to the resolution during

File: packages/askui-nodejs/src/core/model/custom-element-json.ts [hunk 2]
```diff
@@ -48,9 +47,24 @@ export interface CustomElementJson {
    * probably not what you want) and `1.0` (= elements need to look exactly
    * like defined by `CustomElementJson` which is unlikely to be achieved
    * as even minor differences count). Defaults to `0.9`.
+   *
+   * **Important**: The `threshold` impacts the prediction quality.
    */
   threshold?: number | undefined,
 
+  /**
+   * A threshold for when to stop searching for UI elements similar to the
```
Comment (JohannesDienst-askui) [inline packages/askui-nodejs/src/core/model/custom-element-json.ts:56]: I like this explanation. Understood it with the first read!

File: packages/askui-nodejs/src/core/model/custom-element-json.ts [hunk 3]
```diff
@@ -67,10 +81,14 @@ export interface CustomElementJson {
    * only the green ones captured by `customImage` are filtered for using 'RGB'.
    * Defaults to 'grayscale'.
    *
-   * **Important**: This increases the prediction time quite a bit. So only use
-   * it when absolutely necessary.
+   * **Important**: The `imageCompareFormat` impacts the prediction time as
+   * well as quality. Although this highly depends on our use case, the other
```
Comment (JohannesDienst-askui) [inline packages/askui-nodejs/src/core/model/custom-element-json.ts:85]: I wonder if we can remove the addressing of the user as we do not do it in the rest of the docs.

```suggestion
   * well as quality. Although this highly depends on the use case, the other
```

**Review by JohannesDienst-askui — CHANGES_REQUESTED** [review]
I like the descriptions. Just some minor things.

File: packages/askui-nodejs/src/core/model/custom-element-json.ts (hunk 1 above)
Comment (adi-wan-askui) [reply packages/askui-nodejs/src/core/model/custom-element-json.ts:12]: What do you mean by that?

Comment (adi-wan-askui) [reply packages/askui-nodejs/src/core/model/custom-element-json.ts:12]: I think that you meant the wording. I changed it a bit. See if you like it 🙂

Comment (JohannesDienst-askui) [reply packages/askui-nodejs/src/core/model/custom-element-json.ts]: Yes, maybe it was just me but I thought the description was somehow vague 😄 . Great improvement on clarity (imho)!

**Review by JohannesDienst-askui — APPROVED** [review]

**Pull request merged**